rasa/nlu/selectors/response_selector.py

```diff
@@ -434,10 +428,8 @@ def preprocess_train_data(self, training_data: TrainingData) -> RasaModelData:
                 "all retrieval intents."
             )
 
-        label_attribute = RESPONSE if self.use_text_as_label else INTENT_RESPONSE_KEY
-
         label_id_index_mapping = self._label_id_index_mapping(
-            training_data, attribute=label_attribute
+            training_data, attribute=self.label_attribute
         )
```

Comment:
"However, in the future we should avoid changing any code that is unrelated to the original issue" ;) :D  ... yeah, I can do that as well.

Reply:
What's the original reason `label_attribute` was made a `self` variable? i guess because we don't want to pass around the label attribute?

Reply:
yep true - removed that parameter in other functions as well

Reply:
I remember why that was still there. Because self._label_id_index_mapping is pretty generic and could be moved elsewhere (if I had replaced the training_data.intent_messages with just messages). But since that's in there and I don't know where to put this now, I just removed the parameter instead.